### ISSUE TITLE
Bug Body parameter schema is not generated

### ISSUE THREAD

**Issue by denis-pujdak-adm-it** [domaindrivendev/Swashbuckle.AspNetCore#1452]
I am trying to upgrade my `IDocumentFilter` from v4 to v5. Currently I am on `v5.0.0-rc5`.
So before I was using `new BodyParameter() { Schema = schemaRegistry.GetOrRegister(parameterType) }` and it was working fine.
Now I assume I am using the api correctly (as there is no doc about it available) my example is:
```
new OpenApiParameter()
{
  Name = "query",
  Content = { { "application/json", new OpenApiMediaType() { Schema = docFilterContext.SchemaGenerator.GenerateSchema(parameterType, docFilterContext.SchemaRepository) } }, },
}
```
If I run my website there is a parameter shown as `object` with no properties.

I think it is a bug because my `response` model is rendered properly.
If this is not a bug but a wrong implementation could we add some manual how to for example write a new `post` and `get` operations from scratch using `OpenApiPathItem`.

Very appreciate any help.

![image](https://user-images.githubusercontent.com/13495631/71823176-8a2e4980-308e-11ea-9238-8b2060061e0a.png)

**Comment by denis-pujdak-adm-it** [comment]
Switched to v2 of open api. Same issue.
```
app.UseSwagger(c =>
            {
                ...
                c.SerializeAsV2 = true;
            });
```

**Comment by denis-pujdak-adm-it** [comment]
Tried `requestBody` instead. Still same issue. It is not generating a schema:
```
            var x = new OpenApiPathItem();
            x.AddOperation(OperationType.Post, new OpenApiOperation()
            {
                Tags = new[] { new OpenApiTag() { Name = dataRouteOption.Service } },
                OperationId = $"{dataRouteOption.Operation}",
                RequestBody = dataRouteOption.Parameters.Any() ? new OpenApiRequestBody()
                {
                    Required = true,
                    Content = {
                        { "application/json", new OpenApiMediaType() { Schema = GetOrAddTypeSchema(filterContext, dataRouteOption.Parameters.Values.First()) } },
                        { "application/xml", new OpenApiMediaType() { Schema = GetOrAddTypeSchema(filterContext, dataRouteOption.Parameters.Values.First()) } },
                        { "application/x-www-form-urlencoded", new OpenApiMediaType() { Schema = GetOrAddTypeSchema(filterContext, dataRouteOption.Parameters.Values.First()) } },
                    },
                } : null,
```

The result is:
```
"paths": {
    "/Data/Addresses/GetAddress": {
      "post": {
        "tags": [
          "Addresses"
        ],
        "operationId": "GetAddress",
        "consumes": [
          "application/json",
          "application/xml",
          "application/x-www-form-urlencoded"
        ],
        "produces": [
          "application/json",
          "application/xml"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/AddressFormVm"
            }
          }
        },
        "security": [
          { }
        ]
      }
    },
```

**Comment by domaindrivendev** [comment]
There must be something wrong in you implementation because the following works just fine for me:

```csharp
internal class TestDocFilter : IDocumentFilter
{
    public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
    {
        swaggerDoc.Paths.Add("/test", new OpenApiPathItem
        {
            Operations = new Dictionary<OperationType, OpenApiOperation>
            {
                [OperationType.Post] = new OpenApiOperation
                {
                    Tags = new[] { new OpenApiTag { Name = "test" } },
                    RequestBody = new OpenApiRequestBody
                    {
                        Content = new Dictionary<string, OpenApiMediaType>
                        {
                            ["application/json"] = new OpenApiMediaType
                            {
                                Schema = context.SchemaGenerator.GenerateSchema(typeof(Product), context.SchemaRepository)
                            }
                        }
                    }
                }
            }
        });
    }
}
```

**Comment by domaindrivendev** [comment]
```
"/test:" {
  "post:" {
    "tags": [
      "test"
    ],
    "consumes": [
      "application/json"
    ],
    "parameters": [
    {
      "in": "body",
      "name": "body",
      "schema": {
        "$ref": "#/definitions/Product"
      },
    }
    responses: { },
  }
},
```

**Comment by denis-pujdak-adm-it** [comment]
@domaindrivendev try to add one more media type. You will see it stops working.

**Comment by domaindrivendev** [comment]
Adding the additional media type (e.g. "application/xml") worked just fine too.

To move forward on this, I suggest you create a minimal project from scratch that repro's your issue and then post your project up to Github. Without a repro, I can't do much to help you.

**Comment by denis-pujdak-adm-it** [comment]
Ok. I have found an issue.
The following setting:
```
app.UseSwagger(c =>
            {
                c.RouteTemplate = "api/{documentName}/swagger.json";
                c.SerializeAsV2 = true;
            });
```
makes the request body disappear.

**Comment by domaindrivendev** [comment]
It's not specifically this setting. As I've shown [above](https://github.com/OAI/OpenAPI-Specification/issues/1368), in my example the requestBody is correctly tranlated into a parameter with `name = body` when I use the `SerializeAsV2` setting. That is, it doesn't _disappear_, it's just translated into a `parameter` because that's how the body is described in the v2 format.

If you want me to invistigate this any further, please create a minimal application from scratch that clearly repro's your issue and post it to github so I can pull it down and repro the issue for myself.

**Comment by denis-pujdak-adm-it** [comment]
>> Thanks @domaindrivendev , I got the sample. I am going to create a repo for that and will be back soon today.

**Comment by denis-pujdak-adm-it** [comment]
Ok You got it https://github.com/denis-pujdak-adm-it/Swashbuckle_v3_bug
Thanks for your help.

**Comment by domaindrivendev** [comment]
Thanks for putting this together - I finally got a chance to take a look and was able to see the issue repro. I've narrowed it down to a combination of the `SerializeAsV2` setting AND the addition of one or more _form-based_ media types (i.e. "x-www-form-url-encoded" OR "multipart/form-data") in your operation description.

When the `SerializeAsV2` setting is applied, Swashbuckle delegates to the [OpenAPI.NET library](https://github.com/microsoft/OpenAPI.NET) to convert the V3 spec back into the V2 format. While this functinality has been very useful for providing backwards compatability, I have noticed a couple of issues with the spec conversion feature, and I believe this is yet another one of those.

To be fair, it's a complicated conversion and in V2 _form-based_ payloads are described with `form` parameters as opposed to a single `body` parameter and this distinction is likely what's causing the issue.

Long story short - this will need to be fixed in the upstream library before rather than Swashbuckle and so I would encourage creating a post there. Alternatively, is there a good reason to revert back to the V2 spec. If not, I would suggest you stick with V3 as this doesn't require the reverse conversion and there is far less likely to have issues

**Comment by fairking** [comment]
Closing as I am using v3 since then.
